### ISSUE TITLE
open in the unique qf window will report error

### ISSUE THREAD

**Issue by SolaWing** [kevinhwang91/nvim-bqf#107]
### Neovim version (nvim -v | head -n1)

NVIM v0.8.3

### Operating system/version

macOS 12.6

### How to reproduce the issue

1. open quickfix
2. <C-w>o to close other window
3. <CR> or o on item
4. report errors
<img width="998" alt="图片" src="https://user-images.githubusercontent.com/3897953/224036536-4aede366-8571-4a1b-9793-62a38fca5d6e.png">

...

### Expected behavior

no error, automatically open split buffer or replace buffer

### Actual behavior

report window invalid

**Comment by kevinhwang91** [comment]
Yes, this issue has been known since I dev bqf. I just don't understand why user keep an orphan qf window.

**Comment by SolaWing** [comment]
for myself:
1. sometimes want a full screen to see more items.
2. sometimes by random close the last check buffer, even before realize it, and left an orphan qf window.
